File: src/main/java/ar/edu/um/haberes/rest/model/LegajoCategoriaImputacion.java
/**
 * 
 */
package ar.edu.um.haberes.rest.model;

import java.io.Serializable;
import java.math.BigDecimal;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * @author daniel
 *
 */
@Entity
@EqualsAndHashCode(callSuper = false)
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class LegajoCategoriaImputacion extends Auditable implements Serializable {
	/**
	* 
	*/
	private static final long serialVersionUID = -2131284361342741528L;

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long legajoCategoriaImputacionId;

	private Long legajoId;
	private Integer anho = 0;
	private Integer mes = 0;
	private Integer dependenciaId;
	private Integer facultadId;
	private Integer geograficaId;
	private Integer categoriaId;
	private BigDecimal cuentaSueldos;
	private BigDecimal basico = BigDecimal.ZERO;
	private BigDecimal antiguedad = BigDecimal.ZERO;
	private BigDecimal cuentaAportes;
	
	@OneToOne
	@JoinColumn(name = "legajoId", insertable = false, updatable = false)
	private Persona persona;

	@OneToOne
	@JoinColumn(name = "dependenciaId", insertable = false, updatable = false)
	private Dependencia dependencia;

	@OneToOne
	@JoinColumn(name = "facultadId", insertable = false, updatable = false)
	private Facultad facultad;

	@OneToOne
	@JoinColumn(name = "geograficaId", insertable = false, updatable = false)
	private Geografica geografica;

	@OneToOne
	@JoinColumn(name = "categoriaId", insertable = false, updatable = false)
	private Categoria categoria;

}

File: src/main/java/um/haberes/core/kotlin/model/LegajoCategoriaImputacion.kt
package um.haberes.core.kotlin.model

import jakarta.persistence.*
import java.math.BigDecimal

@Entity
@Table
data class LegajoCategoriaImputacion(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var legajoCategoriaImputacionId: Long? = null,
    var legajoId: Long? = null,
    var anho: Int = 0,
    var mes: Int = 0,
    var dependenciaId: Int? = null,
    var facultadId: Int? = null,
    var geograficaId: Int? = null,
    var categoriaId: Int? = null,
    var cuentaSueldos: BigDecimal? = null,
    var basico: BigDecimal? = BigDecimal.ZERO,
    var antiguedad: BigDecimal = BigDecimal.ZERO,
    var cuentaAportes: BigDecimal? = null,

    @OneToOne
    @JoinColumn(name = "legajoId", insertable = false, updatable = false)
    var persona: Persona? = null,

    @OneToOne
    @JoinColumn(name = "dependenciaId", insertable = false, updatable = false)
    var dependencia: Dependencia? = null,

    @OneToOne
    @JoinColumn(name = "facultadId", insertable = false, updatable = false)
    var facultad: Facultad? = null,

    @OneToOne
    @JoinColumn(name = "geograficaId", insertable = false, updatable = false)
    var geografica: Geografica? = null,

    @OneToOne
    @JoinColumn(name = "categoriaId", insertable = false, updatable = false)
    var categoria: Categoria? = null

) : Auditable()
